packages/contracts-bedrock/src/cannon/interfaces/IPreimageOracle.sol

```diff
@@ -1,41 +1,89 @@
+/// @notice Interface for the PreimageOracle contract.
+interface IPreimageOracle is ISemver {
+    /// @notice A raw leaf of the large preimage proposal merkle tree.
+    struct Leaf {
+        /// @notice The input absorbed for the block, exactly 136 bytes.
+        bytes input;
+        /// @notice The index of the block in the absorption process.
+        uint256 index;
+        /// @notice The hash of the internal state after absorbing the input.
+        bytes32 stateCommitment;
+    }
 
-    /// @notice Reads a preimage from the oracle.
-    /// @param _key The key of the preimage to read.
-    /// @param _offset The offset of the preimage to read.
-    /// @return dat_ The preimage data.
-    /// @return datLen_ The length of the preimage data.
-    function readPreimage(bytes32 _key, uint256 _offset) external view returns (bytes32 dat_, uint256 datLen_);
+    /// @notice Unpacked keys for large preimage proposals.
+    struct LargePreimageProposalKeys {
+        /// @notice The claimant of the large preimage proposal.
+        address claimant;
+        /// @notice The UUID of the large preimage proposal.
+        uint256 uuid;
+    }
+
+    error ActiveProposal();
+    error AlreadyFinalized();
+    error AlreadyInitialized();
+    error BadProposal();
+    error BondTransferFailed();
+    error InsufficientBond();
+    error InvalidInputSize();
+    error InvalidPreimage();
+    error InvalidProof();
+    error NotEOA();
+    error NotInitialized();
+    error PartOffsetOOB();
+    error PostStateMatches();
+    error StatesNotContiguous();
+    error TreeSizeOverflow();
+    error WrongStartingBlock();
 
-    /// @notice Loads of local data part into the preimage oracle.
-    /// @param _ident The identifier of the local data.
-    /// @param _localContext The local key context for the preimage oracle. Optionally, can be set as a constant
-    ///                      if the caller only requires one set of local keys.
-    /// @param _word The local data word.
-    /// @param _size The number of bytes in `_word` to load.
-    /// @param _partOffset The offset of the local data part to write to the oracle.
-    /// @dev The local data parts are loaded into the preimage oracle under the context
-    ///      of the caller - no other account can write to the caller's context
-    ///      specific data.
-    ///
-    ///      There are 5 local data identifiers:
-    ///      ┌────────────┬────────────────────────┐
-    ///      │ Identifier │      Data              │
-    ///      ├────────────┼────────────────────────┤
-    ///      │          1 │ L1 Head Hash (bytes32) │
-    ///      │          2 │ Output Root (bytes32)  │
-    ///      │          3 │ Root Claim (bytes32)   │
-    ///      │          4 │ L2 Block Number (u64)  │
-    ///      │          5 │ Chain ID (u64)         │
-    ///      └────────────┴────────────────────────┘
+    function KECCAK_TREE_DEPTH() external view returns (uint256);
```

Comment:
Out of scope for this PR, but we should not be exposing ALL_CAPS vars. Usually weadd a `camelCase()` getter method that exposes the constants/immutables, because we don't want implementation details (naming convention for constants/immutables) leaking into the ABI

Reply:
DisputeGameV2 will fix